--- build_ios_libs.py.orig	2017-12-11 22:17:46.000000000 +0900
+++ build_ios_libs.py	2017-12-11 22:18:11.000000000 +0900
@@ -63,6 +63,8 @@
       help='Specifies a revision number to embed if building the framework.')
   parser.add_argument('-e', '--bitcode', action='store_true', default=False,
       help='Compile with bitcode.')
+  parser.add_argument('--vp9', action='store_true', default=False,
+      help='Enables VP9.')
   parser.add_argument('--verbose', action='store_true', default=False,
       help='Debug logging.')
   parser.add_argument('--use-goma', action='store_true', default=False,
@@ -176,7 +178,7 @@
   # Build all architectures.
   for arch in architectures:
     BuildWebRTC(args.output_dir, arch, args.build_config, gn_target_name,
-                IOS_DEPLOYMENT_TARGET, LIBVPX_BUILD_VP9, args.bitcode,
+                IOS_DEPLOYMENT_TARGET, args.vp9, args.bitcode,
                 args.use_goma, gn_args, args.build_type == 'static_only')
 
   # Create FAT archive.
