diff --git a/webrtc/tools/BUILD.gn b/webrtc/tools/BUILD.gn
index aa9a5687d0..fb034414cd 100644
--- a/webrtc/tools/BUILD.gn
+++ b/webrtc/tools/BUILD.gn
@@ -218,7 +218,7 @@ if (rtc_enable_protobuf) {
       "../modules/congestion_controller",
       "../modules/rtp_rtcp",
       "../system_wrappers:system_wrappers_default",
-      "//build/config:exe_and_shlib_deps",
+      #"//build/config:exe_and_shlib_deps",
     ]
     public_deps = [
       ":chart_proto",
